images/Dockerfile

```diff
@@ -1,36 +1,34 @@
+ARG DOCKER_VERSION=27.1.1-cli
+ARG RUNNER_CONTAINER_HOOKS_VERSION=0.6.1
+
+# Create a versioned alias for docker:cli
+FROM docker:${DOCKER_VERSION} AS docker-cli
+
 # Source: https://github.com/dotnet/dotnet-docker
-FROM mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy as build
+FROM mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy AS build
 
 ARG TARGETOS
 ARG TARGETARCH
 ARG RUNNER_VERSION
-ARG RUNNER_CONTAINER_HOOKS_VERSION=0.6.1
-ARG DOCKER_VERSION=27.1.1
-ARG BUILDX_VERSION=0.16.2
-
-RUN apt update -y && apt install curl unzip -y
+ARG RUNNER_CONTAINER_HOOKS_VERSION
 
 WORKDIR /actions-runner
-RUN export RUNNER_ARCH=${TARGETARCH} \
-    && if [ "$RUNNER_ARCH" = "amd64" ]; then export RUNNER_ARCH=x64 ; fi \
-    && curl -f -L -o runner.tar.gz https://github.com/actions/runner/releases/download/v${RUNNER_VERSION}/actions-runner-${TARGETOS}-${RUNNER_ARCH}-${RUNNER_VERSION}.tar.gz \
-    && tar xzf ./runner.tar.gz \
-    && rm runner.tar.gz
 
-RUN curl -f -L -o runner-container-hooks.zip https://github.com/actions/runner-container-hooks/releases/download/v${RUNNER_CONTAINER_HOOKS_VERSION}/actions-runner-hooks-k8s-${RUNNER_CONTAINER_HOOKS_VERSION}.zip \
-    && unzip ./runner-container-hooks.zip -d ./k8s \
-    && rm runner-container-hooks.zip
+# Configure git-core/ppa based on guidance here: https://git-scm.com/download/linux
+# The second `apt-get update` isn't strictly necessary, but let's leave it here to force the
+# image build to break if the ppa is suddenly unavailable.
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends curl unzip sudo lsb-release gpg-agent software-properties-common \
+ && add-apt-repository ppa:git-core/ppa \
+ && apt-get update
 
 RUN export RUNNER_ARCH=${TARGETARCH} \
-    && if [ "$RUNNER_ARCH" = "amd64" ]; then export DOCKER_ARCH=x86_64 ; fi \
-    && if [ "$RUNNER_ARCH" = "arm64" ]; then export DOCKER_ARCH=aarch64 ; fi \
-    && curl -fLo docker.tgz https://download.docker.com/${TARGETOS}/static/stable/${DOCKER_ARCH}/docker-${DOCKER_VERSION}.tgz \
-    && tar zxvf docker.tgz \
-    && rm -rf docker.tgz \
-    && mkdir -p /usr/local/lib/docker/cli-plugins \
-    && curl -fLo /usr/local/lib/docker/cli-plugins/docker-buildx \
-        "https://github.com/docker/buildx/releases/download/v${BUILDX_VERSION}/buildx-v${BUILDX_VERSION}.linux-${TARGETARCH}" \
-    && chmod +x /usr/local/lib/docker/cli-plugins/docker-buildx
+ && if [ "$RUNNER_ARCH" = "amd64" ]; then export RUNNER_ARCH=x64 ; fi \
```

Comment:
why did you remove the arm64 condition?

Reply:
Didn't intend to. This is why we code review! Added it back. Will push an update.

Reply:
RUNNER_ARCH for arm64 wasn't in the upstream for actions-runner-$TARGETOS-$RUNNER_ARCH-$RUNNER_VERSION.tar.gz. I didn't remove it. The upstream link for that uses x64 and arm64. The former is set by the conditional, the latter is the same as TARGETARCH.

Reply:
FWIW...the diff tool isn't doing us any favors. Look at the whole file in context and you will see what I'm talking about. It's diffing the docker cli install and actions-runner install here, so it looks like I remove it.

Reply:
I see it now – thanks for taking the time to explain it! That is a strange coincidence.